stats.cpp

```diff
@@ -1,25 +1,37 @@
 #include "stats.h"
-#include <limits>
+#include <limits>   // for quiet_NaN
+#include <cfloat>   // for FLT_MAX (min and max float)
 
-Stats Statistics::ComputeStatistics(const std::vector<double>& statsVec) {
-    //Implement statistics here
-    if (statsVec.empty()) {
- 		   double nan = std::numeric_limits<double>::quiet_NaN();
- 		   return Statistics::Stats(nan, nan, nan);
- 	}
- 	
- 	double sum = 0.0;
- 	double minVal = statsVec[0];
- 	double maxVal = statsVec[0];
- 
- 	for (int i=0; i<statsVec.size(); i++) {
- 		sum += statsVec[i];
- 		if (statsVec[i] < minVal) 
-            minVal = statsVec[i];
- 		if (statsVec[i] > maxVal) 
-            maxVal = statsVec[i];
- 	}
- 
- 	double avg = sum / statsVec.size();
- 	return Statistics::Stats(avg, minVal, maxVal);
-}
+namespace Statistics {
+
+    Stats ComputeStatistics(const std::vector<float>& numbers) {
+        Stats result;
+        
+        if (numbers.empty()) {
+            result.average = std::numeric_limits<float>::quiet_NaN();
```

Comment:
If NaN is a way to express "not valid" values, there can be input numbers that are NaN as well. Think of what should be the behavior in that case. What questions do you need to ask, in order to ascertain that?